### ISSUE TITLE
Add "git blame" function to summarize package authorship

### ISSUE THREAD

**Issue by ablack3** [darwin-eu-dev/PaRe#20]
I wrote a function that summarizes package authorship and used it to analyze authorship for Hades, Atlas, and all repos in github.com/ohdsi-studies.

Would you like to add this as an optional utility function to PaRe? It requires the source code in a git repo to run.

Basic idea of the function:
```
# Count lines of source code in a git repository by filename and last author
# 
# @param dir The path to a folder that is a git repo containing source code
# @param regex A regular expression used to select files in the git repo for 
# @value A dataframe with package name, file name, author name, and lines of code
gitBlameRepo <- function(dir = here::here(), regex = ".R|.r") {

  #TODO check that git is installed and "dir" folder containing a git repo
  require(dplyr)
  
  f <- withr::with_dir(dir, list.files(".", pattern = regex, full.names = TRUE, recursive = TRUE))

  df <- dplyr::tibble(package = dir, file = f) %>% 
    dplyr::mutate(author = purrr::map(file, ~withr::with_dir(here::here("atlas_repos", i), 
                            system(glue::glue("git blame --line-porcelain {.x} | sed -n 's/^author //p' | sort | uniq -c | sort -rn"), intern = T)) %>% 
                                 stringr::str_trim()))
  
  df %>% 
    tidyr::unnest(cols = "author") %>% 
    dplyr::mutate(n = stringr::str_extract(author, "^\\d+")) %>% 
    dplyr::mutate(author2 = stringr::str_remove(author, "^\\d+ ")) %>% 
    dplyr::select(package, file, author = author2, n)
}
```

**Comment by edward-burn** [comment]
+1 from me, this would be really nice to have at the top of the report

**Comment by mvankessel-EMC** [comment]
@edward-burn @ablack3, after the recent update, to use `R6` objects to represent the repository, files, and functions, I've made a first version using `git2r::blame`. `gitBlame()` is a method of the `File` object, that is run when initialized, and writes a `tibble` to the private field in the `File` object. The `tibble` looks like this for the file **benchmarkIncidencePrevalence.R** of IncidencePrevalence:

```
# A tibble: 125 × 2
   author        file                          
   <chr>         <chr>                         
 1 edward-burn   benchmarkIncidencePrevalence.R
 2 edward-burn   benchmarkIncidencePrevalence.R
 3 edward-burn   benchmarkIncidencePrevalence.R
 4 edward-burn   benchmarkIncidencePrevalence.R
 5 edward-burn   benchmarkIncidencePrevalence.R
 6 KimLopezGuell benchmarkIncidencePrevalence.R
 7 edward-burn   benchmarkIncidencePrevalence.R
 8 edward-burn   benchmarkIncidencePrevalence.R
 9 edward-burn   benchmarkIncidencePrevalence.R
10 edward-burn   benchmarkIncidencePrevalence.R
# … with 115 more rows
# ℹ Use `print(n = ...)` to see more rows
```

These tables can be merged to get an overview of the relevant files. Currently the following types of files are supported: _R_, _sql_, _cpp_ (inculding header and object files), and _java_.

Example code to check this out:
```
# Install blame branch
remotes::install_github("darwin-eu-dev/PaRe@blame)

# Make Repository object
pathToPkg <- "../IncidencePrevalence/"
repo <- PaRe::Repository$new(pathToPkg)

# get first R file
files <- repo$getFiles()
file <- files$R[[1]]

# get file info
file$getBlameTable()
```

**Comment by ablack3** [comment]
Nice! So I was thinking the data frame would have repo_name, username, filename, and number_of_lines. It would be a summary table for a whole repo.

**Comment by mvankessel-EMC** [comment]
Included in version 0.1.7 294bd2278ee1906d6a0768dcc545e855986ce79c
